.readthedocs.yaml

```diff
@@ -0,0 +1,64 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2024, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+
+# Read the Docs configuration file for the fparser documentation
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details.
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.12"
+  apt_packages:
+    - graphviz
+
+# Build documentation in the doc/source directory with Sphinx
+sphinx:
+   configuration: doc/source/conf.py
+
+# If using Sphinx, optionally build your docs in additional formats such as PDF
+# formats:
+#    - pdf
+
+# Optionally declare the Python requirements required to build your docs.
+python:
+   install:
+     - requirements: doc/pip_requirements.txt
```

Comment:
The rest of the PR looks good, but I am just wondering about this line.

At least all the dependencies listed on doc/pip_requirements.txt should also be in the extras_require doc section of the setup.cfg since in the local environment I would expect `pip install -e .[doc]` to install everything.

But then, is this duplication necessary? Wouldn't the readthedocs just work fine without this line and file if all the dependencies are properly listed in the setup.cfg configuration?

And finally, (maybe not this PR but since we are at it) why do we still have a setup.cfg if we already have a pyproject.toml. I thought this file was to centralise all the configurations as shown in https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

Reply:
I think I must have only added the bare minimum to get setuptools_scm working. I've migrated the rest of setup.cfg over to it now and removed that file.

Reply:
I've also now removed pip_requirements (and moved the packages it listed under the `doc` target in pyproject.toml). The docs still build fine.